hummingbird.py

```diff
@@ -183,9 +183,37 @@ def process_data(self, data):
     if self.start_time is None:
       self.start_time = data.start_time
 
+  def max_of_filtered_arr(self, data, threshold=1):
+    """Remove glitch from data segment.
+
+    Perform De-glitch per 0.1us segment.
+    Remove voltage difference between two sample points larger
+    than 1V.
+
+    Args:
+      data: raw voltage data
+      threshold: difference larger than threshold would be removed
+                 (default: 1V)
+
+    Returns:
+      maxx: the maxium voltage of the filtered data
+    """
+    data = np.array(data.copy())
+    length = round(1e-7 / self.sampling_period)
+    segments = len(data) // length
+    for i in range(segments):
+      arr = data[i*length:(i+1)*length]
+      difference = np.abs(arr - np.median(arr))
+      mask = difference > threshold
+      arr[mask] = np.median(arr)
+      data[i*length:(i+1)*length] = arr
```

Comment:
Since you are calculating the "maximum value that is less than median + 1",
something like this should work: `np.max(arr[arr < median + threshold])`.